### ISSUE TITLE
add 'status' as a sub statement to 'module'

### ISSUE THREAD

**Issue by kwatsen** [netmod-wg/yang-next#28]
So that an entire module (routing-cfg) can be deprecated w/o having to deprecate every node.

**Comment by llhotka** [comment]
Does it make sense? It should suffice to mention it in module description. If somebody needs to add such a module to the data model, he must have a reason to do so, i.e. some data on the server. It could be useful though to have status as a part of IANA registration.

**Comment by kwatsen** [comment]
the modules are extracted from the draft.  we could just rely on 'description' statements, but that's kind of weak compared to using 'status' statements.

**Comment by abierman** [comment]
I like the module status idea, except it should have different syntax.
It would be useful to have a "redirect" to the new module name if
the entire module is deprecated or obsolete.

**Comment by mjethanandani** [comment]
I agree with Andy. For something that is being obsoleted/replaced, in addition to the status, it is always helpful to know where to get information for what replaces it.

**Comment by schoenw** [comment]
I do not think that status is inherited and the editing effort to add status whatever is not that much (and if it is too much, write an editor macro or a tool that generates the status statements).

**Comment by mbj4668** [comment]
Even if the status is explicit in all definitions in a module, I think it would be useful to allow status on the top level.

**Comment by BalazsLengyel** [comment]
Support this.
IMHO status must be inherited.  If a list is obsolete (is probably not implemented) and a  leaf within it is current, it is still (probably) unavailable as the list is not implemented.

**Comment by rgwilton** [comment]
It would be useful to be able to make an entire module as being "status experimental" as another justification as to why this would be useful.

**Comment by kwatsen** [comment]
Does yang-versioning work supercede this?   since now NBC changes can* be made.
